data_structures/src/chain.rs

```diff
@@ -369,6 +370,16 @@ impl Transaction {
             let pointed_value = pointed_transaction
                 .get_output_value(index)
                 .ok_or_else(|| TransactionError::OutputNotFound(transaction_id, index))?;
+                */
+            let pointed_value = pool
+                .get(&input.output_pointer())
+                .ok_or_else(|| {
+                    TransactionError::OutputNotFound(
+                        input.output_pointer().transaction_id,
+                        input.output_pointer().output_index as usize,
+                    )
```

Comment:
Why don't we just define `TransactionError::OutputNotFound` as TransactionError::OutputNotFound(OutputPointer)`, then this snippet of code would look like:

```rust
TransactionError::OutputNotFound(input.output_pointer())
```